File: Projekt2/backend/app/schemas/movie.py
from datetime import datetime

from pydantic import BaseModel


class MovieOut(BaseModel):
    id: int
    title: str
    description: str
    genre: str
    created_at: datetime
    average_rating: float | None = None
    rating_count: int = 0

    model_config = {"from_attributes": True}


class MovieCreate(BaseModel):
    title: str
    description: str = ""
    genre: str


class MovieUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    genre: str | None = None


class RatingOut(BaseModel):
    id: int
    movie_id: int
    user_id: int
    score: int
    created_at: datetime

    model_config = {"from_attributes": True}


class RatingCreate(BaseModel):
    score: int

File: Projekt2/frontend/src/types/Movie.ts
export interface Movie {
  id: number;
  title: string;
  description: string;
  genre: string;
  created_at: string;
  average_rating: number | null;
  rating_count: number;
}

export interface MovieCreate {
  title: string;
  description: string;
  genre: string;
}

export interface MovieUpdate {
  title?: string;
  description?: string;
  genre?: string;
}

export interface Rating {
  id: number;
  movie_id: number;
  user_id: number;
  score: number;
  created_at: string;
}

export interface TopMovie {
  id: number;
  title: string;
  description: string;
  genre: string;
  avg: number;
  count: number;
}
